cpp/src/arrow/acero/concurrent_queue_internal.h

```diff
@@ -35,7 +36,12 @@ class ConcurrentQueue {
   //
   T Pop() {
```

Comment:
Shall we just remove the `Pop()` method considering there is no usage of it at all.

Reply:
I think we should leave it to maintain API compatibility between ConcurrentQueue and BackpressureConcurrentQueue, especially since we are considering making it public there might be uses for it.

Reply:
Oh, I mean remove it from both `ConcurrentQueue` and `BackpressureConcurrentQueue` so there isn't compatibility issue between them.

And especially now, before making them public, is last timing that we are free to make such changes. After they are made public, it's much harder to remove them than adding them back (if necessary).

What do you think?

Reply:
Hmmm. I like the idea of removing potential unsafe call. I thought Pop is is used but in fact only TryPop() is used.

If we could eliminate `Front` as well...
Front is unfortunately heavily used in sorted_merge_node and asof_join_node.
Any ideas?

Reply:
Well, my suggestion of removing `Pop` is solely because it is not called anywhere (or in other words, replaced by `WaitAndPop`), not that it is "unsafe". And this reason doesn't apply to `Front`, so I wouldn't suggest removing it considering it is being widely used.

Reply:
Understood. Thank you for the reviews.